Sources/PointFree/Episode/Show.swift

```diff
@@ -499,5 +709,90 @@ func unsafeMark(from markdown: String) -> String {
   return String(cString: cString)
 }
 
+private func isEpisodeViewable(
+  _ permission: EpisodePermission,
+  _ episode: Episode,
+  _ subscriptionStatus: Stripe.Subscription.Status?
+  ) -> Bool {
+
+  return !episode.subscriberOnly || subscriptionStatus == .some(.active)
+}
+
+private func isEpisodeViewable(for permission: EpisodePermission) -> Bool {
+  switch permission {
+  case .loggedIn(_, .isSubscriber):
+    return true
+  case .loggedIn(_, .isNotSubscriber(.isCredit)):
+    return true
+  case let .loggedIn(_, .isNotSubscriber(.isNotCredit(isSubscriberOnly))):
+    return !isSubscriberOnly
+  case let .loggedOut(isSubscriberOnly):
+    return !isSubscriberOnly
+  }
+}
+
+private func isSubscribeBannerVisible(for permission: EpisodePermission) -> Bool {
+  switch permission {
+  case .loggedIn(_, .isSubscriber):
+    return false
+  case .loggedIn(_, _), .loggedOut(_):
+    return true
+  }
+}
+
+private enum EpisodePermission {
```

Comment:
this is how I model what kind of permission a user has for an episode. we wanna be able to message depending on if you are logged in/out, subscriber or not, and have used a credit or not.

Reply:
V cool.